robots/car_display.h

```diff
@@ -107,6 +107,64 @@ class CarDisplay
         return out;
     }
 
+    std::vector<float> getMouseClickPixelPosition() const
+    {
+        std::vector<float> out;
+        out.push_back(m_MouseClickPosition[0]);
+        out.push_back(m_MouseClickPosition[1]);
+        return out;
+    }
+
+    //! draws the rectangle at the desired coordinates
+    void drawRectangleAtCoordinates(const SDL_Rect &rectangle)
+    {
+        SDL_SetRenderDrawColor(m_Renderer, 0, 0, 255, 255);
+        SDL_RenderFillRect(m_Renderer, &rectangle);
+    }
+
+    //! return the current SDL_Renderer pointer if we want to draw additional graphics
+    //! elements to the screen.
+    SDL_Renderer* getRenderer() {
+        return m_Renderer;
+    }
+
+    //! clears the screen to the selected color. Call this first, before draw()
+    void clearScreen() {
+        // Clear the entire screen to our selected color.
+        SDL_RenderClear(m_Renderer);
+    }
+
+    //! changes pixel to millimeter
+    void pixelToMM(const int x, const int y, millimeter_t &xMM, millimeter_t &yMM) const
+    {
+        xMM = (x - (WindowWidth / 2)) * m_MMPerPixel;
+        yMM = -(y - (WindowHeight / 2)) * m_MMPerPixel;
+    }
+
+    //! change from millimeter to pixel
+    void mmToPixel(const millimeter_t x, const millimeter_t y, int &xPixel, int &yPixel) const
+    {
+        xPixel = static_cast<int>(x / m_MMPerPixel) + (WindowWidth / 2);
+        yPixel = -static_cast<int>(y / m_MMPerPixel) + (WindowHeight / 2);
+    }
+
+    //! draw lines between a list of points 
+    void drawLinesBetweenRects(std::vector<SDL_Rect> listRects, SDL_Renderer *renderer) {
```

Comment:
I'm not sure how general the functionality to draw lines between the top-left corner of rectangles is :smile: maybe move this into the example? Also the rects should be passed as a reference i.e. ``const std::vector<SDL_Rect> &listRects`` so they don't get copied every time its called